vm/src/iterator.rs

```diff
@@ -66,20 +65,19 @@ pub fn length_hint(vm: &VirtualMachine, iter: PyObjectRef) -> PyResult<Option<us
             }
         }
     };
-    let result = result
+    match result
         .payload_if_subclass::<PyInt>(vm)
         .ok_or_else(|| {
             vm.new_type_error(format!(
                 "'{}' object cannot be interpreted as an integer",
                 result.class().name()
             ))
         })?
-        .as_bigint();
-    if result.is_negative() {
-        return Err(vm.new_value_error("__length_hint__() should return >= 0".to_owned()));
+        .try_to_primitive(vm)
+    {
+        Ok(hint) => Ok(Some(hint)),
+        Err(_) => Err(vm.new_value_error("__length_hint__() should return >= 0".to_owned())),
```

Comment:
The original intend looks like distinguising overflow error and negative value error

Reply:
ah, good point. I'll address that.